app/src/main/java/com/globant/practice/presentation/presenter/SubscriberDetailsPresenter.java

```diff
@@ -124,26 +120,21 @@ public SubscriberDetailsPresenter(FetchSubscriberProfileInteractor profileIntera
      */
     public void fetchSubscriberDetails(String login) {
         subscriberDetailsState.setLogin(login);
-        if (subscriberDetailsState.isLoading() && subscriberProfile != null && subscriberRepositories != null) {
-            subscriberDetailsState.setError(null);
-            subscriberDetailsState.setProfile(subscriberProfile);
-            subscriberDetailsState.setSubscriberRepositories(subscriberRepositories);
-            subscriberDetailsState.setLoading(false);
-            if (isViewAttached()) {
-                view.render(subscriberDetailsState);
-            }
-        } else {
+        if (!subscriberDetailsState.isLoading() && subscriberDetailsState.getProfile() == null && subscriberDetailsState.getSubscriberRepositories() == null) {
             subscriberDetailsState.setError(null);
             subscriberDetailsState.setProfile(null);
             subscriberDetailsState.setSubscriberRepositories(null);
             subscriberDetailsState.setLoading(true);
-            if (isViewAttached()) {
-                view.render(subscriberDetailsState);
-                subscriberProfileObservable = profileInteractor.execute(subscriberDetailsState.getLogin());
-                subscriberProfileObservable.subscribeOn(Schedulers.newThread())
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe(getSubscriberProfile);
-            }
+            subscriberProfileObservable = profileInteractor.execute(subscriberDetailsState.getLogin());
+            subscriberProfileObservable.subscribeOn(Schedulers.newThread())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(getSubscriberProfile);
+        } else if (subscriberDetailsState.getProfile() != null && subscriberDetailsState.getSubscriberRepositories() != null) {
```

Comment:
Is this necessary?
When the last request (repositories) finishes, you already sets error null and loading false.